lib/smb2/tree.rb

```diff
@@ -172,4 +219,26 @@ def disposition_from_file_mode(mode)
     end
   end
 
+  def class_array_from_blob(blob, klass)
+    class_array = []
+    offset = 0
+
+    loop do
+      length = blob[offset, 4].unpack('V').first
```

Comment:
`unpack` will work on a string longer than the format.  You can do this directly on blob, without the slicing, i.e.:
```ruby
length = blob.unpack('V').first
```